### PR TITLE
Awi/style suggestions 2

### PR DESCRIPTION
**Pull request by awiawi**
Hey,

here are some suggestions for improvements.

### CODE REVIEW

File: 02_tools-and-packages/01_plot-packages.ipynb [hunk 1]
```diff
@@ -96,8 +96,8 @@
    "outputs": [],
    "source": [
     "# First, we create some data to plot, which is not that handy with lists.\n",
-    "x = [i for i in range(-20, 20)]\n",
-    "y = [x[i]**2 for i in range(0,40)]"
+    "x = [i for i in range(-20, 21)]\n",
+    "y = [x_i**2 for x_i in x]"
```
Comment (StephanRhode) [inline 02_tools-and-packages/01_plot-packages.ipynb]: Thanks, this is way more pythonic than my list comprehension.

File: 02_tools-and-packages/02_numpy.ipynb [hunk 2]
```diff
@@ -137,9 +137,9 @@
     "## Create arrays manually\n",
     "\n",
     "* The \"manual\" syntax to create a numpy array is \n",
-    "  * `np.array([x, x])` for one dimensional arrays \n",
-    "  * and `np.array([[x, x], [x, x]])` for multidimensional arrays.\n",
-    "* Add to this, various functions like `np.ones()`, `np.eye()`, `np.arrange()`, `np.linspace()`, and many more create specific arrays which are often required in matrix computing."
+    "  * `np.array(a_list)` for one dimensional arrays \n",
```
Comment (StephanRhode) [inline 02_tools-and-packages/02_numpy.ipynb]:
```suggestion
    "  * `np.array([x, x])` for one dimensional arrays \n",
```

File: 02_tools-and-packages/02_numpy.ipynb [hunk 3]
```diff
@@ -137,9 +137,9 @@
...
+    "  * and `np.array([a_list, another_list])` for multidimensional arrays.\n",
```
Comment (StephanRhode) [inline 02_tools-and-packages/02_numpy.ipynb]: Could we switch back to this style? I guess it is more descriptive for students than `a_list`.

```suggestion
    "  * `np.array([x, x])` for one dimensional arrays\n",
    "  * and `np.array([[x, x], [x, x]])` for multidimensional arrays.\n",
```

File: 02_tools-and-packages/02_numpy.ipynb [hunk 4]
```diff
@@ -908,9 +905,9 @@
     {
      "data": {
       "text/plain": [
-       "array([[1., 1., 1., 1.],\n",
-       "       [1., 1., 1., 1.],\n",
-       "       [1., 1., 1., 1.]])"
+       "array([[ 0,  1,  2,  3],\n",
+       "       [ 4,  5,  6,  7],\n",
+       "       [ 8,  9, 10, 11]])"
```
Comment (StephanRhode) [inline 02_tools-and-packages/02_numpy.ipynb]: Much better example, thanks!

File: 02_tools-and-packages/03_scipy.ipynb [hunk 5]
```diff
@@ -756,11 +756,11 @@
     "\n",
     "$y'' + y' + 4 y = 0$. \n",
     "\n",
-    "To evaluate this second order ode, we need to convert it into a set of first order ode. The trick is to use this substitution: $x_2 = y'$, $x_1 = y$, which yields\n",
+    "To evaluate this second order ode, we need to convert it into a set of first order ode. The trick is to use this substitution: $x_0 = y$, $x_1 = y'$, which yields\n",
```
Comment (StephanRhode) [inline 02_tools-and-packages/03_scipy.ipynb]: Nice fix for indexing starts at zero comment below.

File: 02_tools-and-packages/05_sklearn.ipynb [hunk 6]
```diff
@@ -179,7 +179,7 @@
     }
    },
    "source": [
-    "Always **reserve some data for testing** your algorithm. There are many algorithms which suffer from overfitting. Hence, test the trained algorithm on some unseen data. In practice you can use cross validation to determine the performance of your algorithm."
+    "Always [reserve some data for testing](https://scikit-learn.org/stable/modules/cross_validation.html) your algorithm. There are many algorithms which suffer from overfitting. Hence, test the trained algorithm on some unseen data. In practice you can use cross validation to determine the performance of your algorithm."
```
Comment (StephanRhode) [inline 02_tools-and-packages/05_sklearn.ipynb]: Could we use a bold font here?

```suggestion
    "Always [**reserve some data for testing**](https://scikit-learn.org/stable/modules/cross_validation.html) your algorithm. There are many algorithms which suffer from overfitting. Hence, test the trained algorithm on some unseen data. In practice you can use cross validation to determine the performance of your algorithm."
```

File: 02_tools-and-packages/05_sklearn.ipynb [hunk 7]
```diff
@@ -190,7 +190,7 @@
     }
    },
    "source": [
-    "Consider **feature reduction**, **feature scaling**, and other data clearance tasks before machine learning. It is often more successful to scale and clean the data before you move to more advanced machine learning algorithms.\n",
+    "Consider [feature reduction](https://scikit-learn.org/stable/modules/feature_selection.html), [feature scaling](https://scikit-learn.org/stable/modules/preprocessing.html#standardization-or-mean-removal-and-variance-scaling), and other data clearance tasks before machine learning. It is often more successful to scale and clean the data before you move to more advanced machine learning algorithms.\n",
```
Comment (StephanRhode) [inline 02_tools-and-packages/05_sklearn.ipynb]:
```suggestion
    "Consider [**feature reduction**](https://scikit-learn.org/stable/modules/feature_selection.html), [**feature scaling**](https://scikit-learn.org/stable/modules/preprocessing.html#standardization-or-mean-removal-and-variance-scaling), and other data clearance tasks before machine learning. It is often more successful to scale and clean the data before you move to more advanced machine learning algorithms.\n",
```

File: 02_tools-and-packages/02_numpy.ipynb (hunk 3 above)
Comment (awiawi) [reply 02_tools-and-packages/02_numpy.ipynb]: I was a bit confused by `[x, x]`. I was thinking it was a fixed number of arguments, but this is not the case. I must admit that `a_list` is also not very precise. Maybe you can use `[x_0, x_1, ..., x_n]`?

File: 02_tools-and-packages/05_sklearn.ipynb (hunk 6 above)
Comment (awiawi) [reply 02_tools-and-packages/05_sklearn.ipynb]: Of course. I tried the `**` around the brackets `[ ]` and it didn't work. :-)

File: 02_tools-and-packages/02_numpy.ipynb (hunk 3 above)
Comment (awiawi) [inline 02_tools-and-packages/02_numpy.ipynb]:
```suggestion
    "  * `np.array([x_0, x_1, ..., x_n])` for one dimensional arrays \n",
    "  * and `np.array([[x_0, x_1, ... x_n], [y_0, y_1, ..., y_n]])` for multidimensional arrays.\n",
```

**Pull request merged**